home/management/commands/updatecourses.py

```diff
@@ -0,0 +1,104 @@
+import requests
+from datetime import datetime
+
+from django.core.management import BaseCommand
+from django.db.models import Q
+
+from home.models import Course, Professor, ProfessorCourse
+from home.utils import Semester
+
+class Command(BaseCommand):
+    def __init__(self):
+        super().__init__()
+        self.total_num_new_courses = 0
+        self.total_num_new_professors = 0
+
+    def add_arguments(self, parser):
+        parser.add_argument("semesters", nargs='+')
+
+    def handle(self, *args, **options):
+        t_start = datetime.now()
+        inputted_semesters = [Semester(s).name() for s in options['semesters']]
+        print(f"Inputted Semesters: {', '.join(inputted_semesters)}")
+
+        for semester in options['semesters']:
+            s = Semester(semester)
+            kwargs = {"semester": semester, "per_page": 100, "page": 1}
+            course_data = requests.get("https://api.umd.io/v1/courses", params=kwargs).json()
+
+            if "error_code" in course_data[0].keys():
+                print(f"umd.io doesn't have data for {s.name()}!")
+                continue
+
+            print(f"Working on courses for {s.name()}...")
+
+            while course_data:
+                for umdio_course in course_data:
+                    course = Course.unfiltered.filter(name=umdio_course['course_id']).first()
+                    if not course:
+                        course = Course(
+                            name=umdio_course['course_id'],
+                            department=umdio_course['dept_id'],
+                            course_number=umdio_course['course_id'][4:],
```

Comment:
are we 100% sure department ids are always 4 characters? I think they probably are, but not sure. I can't remember if we're relying on this somewhere else in the codebase already, or if umdio itself is.

Probably worth throwing an `assert umdio_course['dept_id'] == umdio_course['course_id'][:4]` in here regardless.

Reply:
I know the department ids are all 4 characters. See for yourself:
`SELECT * FROM planetterp.home_course WHERE CHAR_LENGTH(department) > 4;`